test/Quake-QIR/basic.qke

```diff
@@ -42,7 +16,17 @@ module {
           return
      }
 
-    func.func @test_func2(){
+// CHECK-LABEL: define void @test_func(i32 
+// CHECK-SAME:                             %[[VAL_0:.*]]) local_unnamed_addr {
+// CHECK:         %[[VAL_1:.*]] = zext i32 %[[VAL_0]] to i64
+// CHECK:         %[[VAL_2:.*]] = tail call %[[VAL_3:.*]]* @__quantum__rt__qubit_allocate_array(i64 %[[VAL_1]])
```

Comment:
Here, opposed to below there are two allocations since one value is not yet known? What does that mean for the allocation bug - i.e. is the allocation unification running after quake synth and inlining and we should end up with a single allocation at the top throughout the entire entrypoint kernel?

Reply:
This tests the first phase of lowering to QIR which does allow for dynamic allocation of qubits.

This QIR (adaptive profile, I believe) subsequently can be lowered to a base profile QIR which removes the dynamic allocations if and only if the number of qubits is known. One can't generate base profile QIR with dynamic allocations of qubits.